src/test/fixtures/fake-symbol-store.js

```diff
@@ -2,33 +2,50 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+//@flow
+import bisection from 'bisection';
+
+import type {
+  LibSymbolicationRequest,
+  AddressResult,
+} from '../../profile-logic/symbol-store';
+
 export class FakeSymbolStore {
-  constructor(symbolTables) {
-    this._symbolTables = {};
-    for (const debugName in symbolTables) {
-      const entries = Array.from(Object.entries(symbolTables[debugName]));
+  _symbolTables: Map<string, {| addrs: Uint32Array, syms: string[] |}>;
+
+  constructor(symbolTables: Map<string, Map<number, string>>) {
+    this._symbolTables = new Map();
+    for (const [debugName, symbolTable] of symbolTables.entries()) {
+      const entries = Array.from(symbolTable.entries());
       entries.sort(([addr1], [addr2]) => addr1 - addr2);
-      this._symbolTables[debugName] = {
+      this._symbolTables.set(debugName, {
         addrs: new Uint32Array(entries.map(([addr]) => addr)),
         syms: entries.map(([, sym]) => sym),
-      };
-    }
-  }
-
-  getFuncAddressTableForLib(lib) {
-    if (lib.debugName in this._symbolTables) {
-      return Promise.resolve(this._symbolTables[lib.debugName].addrs);
+      });
     }
-    return Promise.reject(new Error());
   }
 
-  getSymbolsForAddressesInLib(requestedAddressesIndices, lib) {
-    if (lib.debugName in this._symbolTables) {
-      const syms = this._symbolTables[lib.debugName].syms;
-      return Promise.resolve(
-        requestedAddressesIndices.map(index => syms[index])
-      );
+  async getSymbols(
+    requests: LibSymbolicationRequest[],
+    successCb: (LibSymbolicationRequest, Map<number, AddressResult>) => void,
+    errorCb: (LibSymbolicationRequest, Error) => void
+  ): Promise<void> {
```

Comment:
should probably start this function with an `await Promise.resolve()` so that the callbacks are never called synchronously

Reply:
oh yeah, good call !